core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1664,10 +1686,11 @@ public long processWalBlock(
 
         lastPartitionTimestamp = txWriter.getPartitionTimestampByTimestamp(partitionTimestampHi);
 
+        boolean sucess = true;
         try {
-            final long maxLagRows = getMaxWalSquashRows();
+            final long maxLagRows = getWalMaxLagRows();
```

Comment:
When `walLagRowCount` is 0 this branch is not exercised because of L1817

```java
if (commitToTimestamp < txWriter.getLagMaxTimestamp() && maxLagRows > 0) {
```

I'll add some tests